src/main/java/com/synopsys/integration/alert/provider/polaris/PolarisApiHelper.java

```diff
@@ -138,24 +138,35 @@ public PolarisApiHelper(final IssueService issueService, final RoleAssignmentSer
 
     private Map<String, Integer> mapIssueTypeToCount(final String projectId, final String branchId, final List<IssueV0Resource> queryIssues) throws IntegrationException {
         final Map<String, Integer> issueTypeCounts = new HashMap<>();
+        final Map<String, String> issueTypeMap = new HashMap<>();
         for (final IssueV0Resource queryIssue : queryIssues) {
-            final String issueKey = queryIssue.getAttributes().getIssueKey();
-            final IssueResourcesSingle populatedIssueResource = issueService.getIssueForProjectBranchAndIssueKeyWithDefaultIncluded(projectId, branchId, issueKey);
             final String subTool = queryIssue.getAttributes().getSubTool();
-            final String issueType = issueService
-                                         .getIssueTypeFromPopulatedIssueResources(populatedIssueResource)
-                                         .map(IssueTypeV0Resource::getAttributes)
-                                         .map(IssueTypeV0Attributes::getName)
-                                         .orElse(subTool);
-            if (!issueTypeCounts.containsKey(issueType)) {
-                issueTypeCounts.put(issueType, 0);
+            if (issueTypeMap.containsKey(subTool)) {
+                final String issueType = issueTypeMap.get(subTool);
+                updateIssueTypeCounts(issueTypeCounts, issueType);
+            } else {
+                final String issueKey = queryIssue.getAttributes().getIssueKey();
+                final IssueResourcesSingle populatedIssueResource = issueService.getIssueForProjectBranchAndIssueKeyWithDefaultIncluded(projectId, branchId, issueKey);
+                final String issueType = issueService
+                                             .getIssueTypeFromPopulatedIssueResources(populatedIssueResource)
+                                             .map(IssueTypeV0Resource::getAttributes)
+                                             .map(IssueTypeV0Attributes::getName)
+                                             .orElse(subTool);
+                issueTypeMap.put(subTool, issueType);
+                updateIssueTypeCounts(issueTypeCounts, issueType);
```

Comment:
We would only have to call this once if we moved the issueTypes variable before the if-statement.

Reply:
I updated the code, so we only update the counts in one place